src/storage/src/mut_records.rs

```diff
@@ -197,4 +254,55 @@ mod tests {
 
         Ok(())
     }
+
+    #[test_async]
+    async fn test_write_records_count() -> Result<(), StorageError> {
+        let test_file = temp_dir().join(TEST_FILE_NAMEC);
+        ensure_clean_file(&test_file);
+
+        const NUM_WRITES: u32 = 4;
+        const OFFSET: i64 = 200;
+
+        let options = ConfigOption {
+            base_dir: temp_dir(),
+            segment_max_bytes: 1000,
+            flush_write_count: NUM_WRITES,
+            ..Default::default()
+        };
+        let mut msg_sink = MutFileRecords::create(OFFSET, &options)
+            .await
+            .expect("create");
+
+        let batch = create_batch();
+        let write_size = batch.write_size(0);
+        debug!("write size: {}", write_size); // for now, this is 79 bytes
+        msg_sink.send(create_batch()).await.expect("create");
+        msg_sink.flush().await.expect("create flush"); // ensure the file is created
+
+        let bytes = read_bytes_from_file(&test_file).expect("read bytes");
+        assert_eq!(bytes.len(), write_size, "incorrect size for write");
+
+        let batch = DefaultBatch::decode_from(&mut Cursor::new(bytes), 0)?;
+        assert_eq!(batch.get_header().magic, 2, "check magic");
+        assert_eq!(batch.records.len(), 2);
+        let mut records = batch.records;
+        assert_eq!(records.len(), 2);
+        let record1 = records.remove(0);
+        assert_eq!(record1.value.inner_value(), Some(vec![10, 20]));
+        let record2 = records.remove(0);
+        assert_eq!(record2.value.inner_value(), Some(vec![10, 20]));
+
+        for _ in 1..NUM_WRITES {
```

Comment:
I'll add some discussion in comments around the tests. I did have negative tests setup, but removed it because of the variability in outcome. In practice it's very hit or miss to catch the no-flush case - the test is basically a race against the kernel layers below also flushing or fully completing their writes internally.

Reply:
Is there way to make it reliable test for flush case?

Reply:
That tests are added are the ones I believe should be reliable for the positive flush cases, but I think I should add more explanation along the lines of these comments regarding why there is no negative flush test.  A negative unit test could maybe be verified with some sort of mock object - but I'm not sure how much value there is in that test.

Reply:
Sounds good